scripts/manage_demux_stats.py

```diff
@@ -0,0 +1,319 @@
+#!/usr/bin/env python
+"""
+This file together with manage_demux_stats_thresholds.py performs the "bclconversion" step of LIMS workflow.
+In common tongue, it:
+ 
+Fetches info from the sequencing process (RunID, FCID; derives instrument and data type)
+Assigns (Q30, Clust per Lane) thresholds to the process (workflow step)
+Reformats laneBarcode.html to "demuxstats_FCID.csv" for usage of other applications
+Assigns a lot of info from laneBarcode.html to individual samples of the process (e.g. %PF)
+Flags samples as QC PASSED/FAILED based on thresholds
+
+Written by Isak Sylvin; isak.sylvin@scilifelab.se"""
+
+#Fetched from SciLife repos
+from genologics.lims import Lims
+from genologics.config import BASEURI, USERNAME, PASSWORD
+from genologics.entities import Process
+import flowcell_parser.classes as classes
+from manage_demux_stats_thresholds import Thresholds
+
+#Standard packages
+from shutil import move
+import os 
+import click
+import csv
+import sys
+import logging
+logger = logging.getLogger('demux_logger')
+
+def problem_handler(type, message):
+    if type == "exit":
+        logger.error(message)
+        sys.exit(message)
+    elif type == "warning":
+        logger.warning(message)
+        sys.stderr.write(message)
+    else:
+        logger.info(message)
+
+"""Fetches overarching workflow info"""
+def manipulate_workflow(demux_process):
+    try:
+        demux_container = demux_process.all_inputs()[0].location[0].name    
+    except:
+        problem_handler("exit", "Container name not found")
+        
+    workflow_types = {"MiSeq Run (MiSeq) 4.0":"Reagent Cartridge ID", "Illumina Sequencing (Illumina SBS) 4.0":"Flow Cell ID",
+              "Illumina Sequencing (HiSeq X) 1.0":"Flow Cell ID"}
+    for k,v in workflow_types.items():
+        #Sequencing step is null if it doesn"t exist
+        workflow = lims.get_processes(udf = {v : demux_container}, type = k)
```

Comment:
Relying on udfs to find the corresponding sequencing steps from a demultiplexing step error prone. 
Given our workflows, the inputs from the demultiplexing are the inputs for the sequencing as well. With that, you can use a get_processes with the `inputartifactlimsid`parameter.